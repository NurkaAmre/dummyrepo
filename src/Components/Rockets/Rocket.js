import React from 'react';
import PropTypes from 'prop-types';
import './rockets.css';

const Rocket = ({
  rocketId, rocketImage, rocketName, rocketDescription,
  reserved, handleBooking,
}) => (
  <div className="flu">
    <img
      className="imge"
      src={rocketImage}
      alt={rocketName}
    />
    <div className="text">
      <h3 className="font">{ rocketName }</h3>
      <p className="my-2">
        { reserved ? (
          <span
            className="reserved"
          >
            Reserved
          </span>
        ) : '' }
        { rocketDescription }
      </p>
      <button
        type="button"
        id={rocketId}
        onClick={() => handleBooking(rocketId)}
        className="btn1"
      >
        { reserved ? 'Cancel Reservation' : 'Reserve Rockets' }
      </button>
    </div>
  </div>
);
Rocket.propTypes = {
  rocketId: PropTypes.string.isRequired,
  rocketImage: PropTypes.string.isRequired,
  rocketName: PropTypes.string.isRequired,
  rocketDescription: PropTypes.string.isRequired,
  reserved: PropTypes.bool.isRequired,
  handleBooking: PropTypes.func.isRequired,
};

export default Rocket;
