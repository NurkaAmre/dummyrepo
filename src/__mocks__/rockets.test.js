import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../Components/redux/configureStore';
import Rocket from '../Components/Rockets/Rocket';

describe('Rockets Container component', () => {
  test('Renders Rockets correctly', () => {
    const rockets = render(

      <Provider store={store}>

        <Rocket />

      </Provider>,

    );
    expect(rockets).toMatchSnapshot();
  });
});
