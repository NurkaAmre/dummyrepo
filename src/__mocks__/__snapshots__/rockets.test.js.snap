// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets Container component Renders Rockets correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flu"
      >
        <img
          class="imge"
        />
        <div
          class="text"
        >
          <h3
            class="font"
          />
          <p
            class="my-2"
          />
          <button
            class="middle none font-sans font-bold center uppercase transition-all disabled:opacity-50 disabled:shadow-none disabled:pointer-events-none text-xs text-white shadow-blue-500/20 hover:shadow-lg focus:opacity-[0.85] focus:shadow-none active:opacity-[0.85] active:shadow-none shadow-none bg-red-700 p-2.5 rounded-md border-blue-700 border-2 hover:shadow-blue-500 hover:text-blue-500 hover:bg-white"
            type="button"
          >
            Reserve Rockets
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flu"
    >
      <img
        class="imge"
      />
      <div
        class="text"
      >
        <h3
          class="font"
        />
        <p
          class="my-2"
        />
        <button
          class="middle none font-sans font-bold center uppercase transition-all disabled:opacity-50 disabled:shadow-none disabled:pointer-events-none text-xs text-white shadow-blue-500/20 hover:shadow-lg focus:opacity-[0.85] focus:shadow-none active:opacity-[0.85] active:shadow-none shadow-none bg-red-700 p-2.5 rounded-md border-blue-700 border-2 hover:shadow-blue-500 hover:text-blue-500 hover:bg-white"
          type="button"
        >
          Reserve Rockets
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
